karapace/schema_registry_apis.py

```diff
@@ -1362,3 +1318,56 @@ def no_master_error(self, content_type: str) -> None:
             content_type=content_type,
             status=HTTPStatus.INTERNAL_SERVER_ERROR,
         )
+
+    def _get_new_schema(self, body: JsonObject, content_type: str) -> ValidatedTypedSchema:
+        schema_type = self._validate_schema_type(content_type=content_type, data=body)
+        references = self._validate_references(content_type, schema_type, body)
+        try:
+            references, new_schema_dependencies = self.schema_registry.resolve_references(references)
+            return ValidatedTypedSchema.parse(
+                schema_type=schema_type,
+                schema_str=body["schema"],
+                references=references,
+                dependencies=new_schema_dependencies,
+            )
+        except InvalidSchema:
+            self.r(
+                body={
+                    "error_code": SchemaErrorCodes.INVALID_SCHEMA.value,
+                    "message": f"Invalid {schema_type} schema",
+                },
+                content_type=content_type,
+                status=HTTPStatus.UNPROCESSABLE_ENTITY,
+            )
+
+    def _get_old_schema(self, subject: Subject, version: Version, content_type: str) -> ValidatedTypedSchema:
```

Comment:
```suggestion
    def _get_old_schema(self, subject: Subject, version: Version, content_type: str) -> ParsedTypedSchema:
```
The old schemas are not fully validated, there can be such old schemas in the Karapace database that cannot pass validation anymore with newer and current Avro library version.

Reply:
Good catch, fixed